backend-test/Controllers/CourseControllerTests.hs

```diff
@@ -203,5 +203,7 @@ runCourseInfoTests = map (\(label, state, dept, expected) -> runCourseInfoTest l
 
 
 -- | Test suite for Course Controller Module
-courseControllerTestSuite :: TestTree
-courseControllerTestSuite = testGroup "Course Controller tests" (runRetrieveCourseTests ++ runIndexTests ++ runCourseInfoTests)
+test_courseController :: TestTree
+test_courseController =
+    withResource (do acquireDatabase) releaseDatabase $ \_ ->
```

Comment:
The `do` is unnecessary; you can just use `acquireDatabase`.

However, given that this code is going to be repeated across multiple test groups, I would refactor the `withResource acquireDatabase releaseDatabase $ \_ -> ...` into a single function `withDatabase` that takes a `String` (label) and `[TestTree]` (the test cases).